parsl/executors/high_throughput/executor.py

```diff
@@ -479,38 +506,60 @@ def scale_out(self, blocks=1):
         r = []
         for i in range(blocks):
             if self.provider:
-                block = self.provider.submit(self.launch_cmd, 1, 1)
-                logger.debug("Launched block {}:{}".format(i, block))
-                if not block:
+                external_block_id = len(self.blocks)
+                launch_cmd = self.launch_cmd.format(block_id=external_block_id)
+                internal_block = self.provider.submit(launch_cmd, 1, 1)
+                logger.debug("Launched block {}->{}".format(external_block_id, internal_block))
+                if not internal_block:
                     raise(ScalingFailed(self.provider.label,
                                         "Attempts to provision nodes via provider has failed"))
-                self.blocks.extend([block])
+                r.extend([str(external_block_id)])
+                self.blocks[str(external_block_id)] = internal_block
             else:
                 logger.error("No execution provider available")
                 r = None
         return r
 
-    def scale_in(self, blocks):
+    def scale_in(self, blocks, block_ids=[]):
```

Comment:
This parameter seems weird: either you'd want to specify a number of anonymous blocks to remove, or an explicit list of blocks to remove, but never both, especially if you specify both with conflicting meaning (eg a list of three IDs, and blocks=2).

Perhaps there should be no added parameter here, but the type of `blocks` should now be either an integer number of anonymous blocks to remove or a list of strings, being a list of identified blocks to remove. That better reflects the behaviour, rather than silently ignoring some parameters sometimes.